### PR TITLE
Update metrics handling in applicationsignalsprocessor

### PR DESCRIPTION
**Pull request by bjrara**
# Description of the issue
This PR is to prepare for the runtime metrics release in Application Signals.

# Description of changes
1. Drop metrics if metric doesn't contain `Telemetry.Source` attribute.
2. Try to copy `aws.local.service` from resource attributes if the attribute is not found on metrics. This is only used for runtime metrics.
3. Respect custom metric names.

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
_Describe what tests you have done._

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
4. Run `make lint`

### CODE REVIEW

**Review by mxiamxia — APPROVED** [review]

File: plugins/processors/awsapplicationsignals/internal/prune/metric_pruner.go [hunk 1]
```diff
@@ -21,6 +22,9 @@ func (p *MetricPruner) ShouldBeDropped(attributes pcommon.Map) (bool, error) {
 				return true, errors.New("Metric attribute " + attributeKey + " must contain only ASCII characters.")
 			}
 		}
+		if _, ok := attributes.Get(common.MetricAttributeTelemetrySource); !ok {
```
Comment (lisguo) [inline plugins/processors/awsapplicationsignals/internal/prune/metric_pruner.go:25]: are there any risks in dropping metrics without Telemetry.Source? Are there any edge cases where a customer might emit application signals metrics without this attribute?

File: plugins/processors/awsapplicationsignals/processor.go [hunk 2]
```diff
@@ -137,7 +138,10 @@ func (ap *awsapplicationsignalsprocessor) processMetrics(ctx context.Context, md
 			metrics := ils.Metrics()
 			for k := 0; k < metrics.Len(); k++ {
 				m := metrics.At(k)
-				m.SetName(metricCaser.String(m.Name())) // Ensure metric name is in sentence case
+				// Check if the first letter of the metric name is not capitalized
+				if len(m.Name()) > 0 && !unicode.IsUpper(rune(m.Name()[0])) {
```
Comment (lisguo) [inline plugins/processors/awsapplicationsignals/processor.go:142]: will there be documentation for custom metrics to ensure the first letter is not capitalized?

File: plugins/processors/awsapplicationsignals/internal/prune/metric_pruner.go (hunk 1 above)
Comment (bjrara) [reply plugins/processors/awsapplicationsignals/internal/prune/metric_pruner.go:25]: In the case where customer is using preview SDK, all the metrics will be dropped. Those SDKs are deprecated though. Also even the traffic is sent to logs, Application Signals has stopped processing those logs emitted from preview SDK while using the new CloudWatch agent.

File: plugins/processors/awsapplicationsignals/processor.go (hunk 2 above)
Comment (bjrara) [reply plugins/processors/awsapplicationsignals/processor.go:142]: It's a good question. I personally prefer respecting the original metric name...

I don't know if there will be documentation for naming convention, perhaps we should revisit this logic whether it makes sense to change the letter case.

**Review by sky333999 — APPROVED** [review]

**Pull request merged**